app/actions/UserActions.ts

```diff
@@ -123,44 +113,35 @@ export function updateUser(
     githubUsername,
     linkedinId,
   } = user;
-  return (dispatch: AppDispatch) =>
-    dispatch(
-      callAPI({
-        types: User.UPDATE,
-        endpoint: `/users/${username}/`,
-        method: 'PATCH',
-        body: {
-          username,
-          firstName,
-          lastName,
-          email,
-          phoneNumber,
-          gender,
-          allergies,
-          selectedTheme,
-          isAbakusMember,
-          emailListsEnabled,
-          githubUsername,
-          linkedinId,
-          ...(options.updateProfilePicture
-            ? {
-                profilePicture,
-              }
-            : null),
-        },
-        schema: userSchema,
-        meta: {
-          successMessage: 'Oppdatering av bruker fullført',
-          errorMessage: 'Oppdatering av bruker feilet',
-        },
-      })
-    ).then((action) => {
-      if (!action || !action.payload) return;
-
-      if (!options.noRedirect) {
-        dispatch(push(`/users/${username}`));
-      }
```

Comment:
So this had to be refactored since the navigation should not be a part of the action function (because then you end up with nested dispatches). And yes all places that called this action was updated accordingly.